fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/external/service/serialization/serializer/loan/LoanBusinessEventSerializer.java

```diff
@@ -84,6 +86,11 @@ public <T> ByteBufferSerializable toAvroDTO(BusinessEvent<T> rawEvent) {
         List<LoanInstallmentDelinquencyBucketDataV1> installmentsDelinquencyData = installmentLevelDelinquencyEventProducer
                 .calculateInstallmentLevelDelinquencyData(event.get(), data.getCurrency());
 
+        Set<LoanTermVariations> activeLoanTermVariations = event.get().getActiveLoanTermVariations();
+        if (activeLoanTermVariations != null) {
```

Comment:
Would it make sense to only set the `emiAmountVariations` if there are any entries?

Reply:
Sadly `getActiveLoanTermVariations()` returns null istead of an empty set, when there are no entries.